sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -1418,15 +1418,16 @@ case class GpuCast(
    * with semantically equivalent expressions. This method is overridden to prevent Spark
    * from substituting non-GPU expressions.
    */
-  override def semanticEquals(other: Expression): Boolean = other match {
-    case g: GpuExpression =>
-      if (this == g) {
-        true
-      } else {
-        super.semanticEquals(g)
-      }
-    case _ => false
-  }
+  // TODO SPARK-35742 canonicalize
```

Comment:
fix spacing and do have an internal issue to reference?

Reply:
I added notes about the issues that need filing after this PR